rotkehlchen/accounting/structures/evm_event.py

```diff
@@ -0,0 +1,147 @@
+import json
+import logging
+import typing
+from dataclasses import dataclass
+from typing import Any, Optional
+
+from rotkehlchen.accounting.structures.base import HISTORY_EVENT_DB_TUPLE_READ, HistoryBaseEntry
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_optional
+from rotkehlchen.types import ChecksumEvmAddress
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+EVM_EVENT_FIELDS = tuple[
+    Optional[str],  # protocol
+    Optional[str],  # product
+    Optional[str],  # address
+    Optional[str],  # extra_data
+]
+
+EVM_EVENT_FIELDS_COUNT = len(typing.get_args(EVM_EVENT_FIELDS))
+
+
+EVM_EVENT_DB_TUPLE_READ = tuple[
```

Comment:
Yeah, I also tried, but couldn't find a way to do this. However this can probably be achieved if instead of making `EVM_EVENT_DB_TUPLE_READ` a long tuple I make it a tuple of 2 subtuples -- one for history base entry and one for evem event.